v3/src/components/graph/components/graph-drop-hint.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useLayoutEffect, useRef, useState} from "react"
+import { AxisPlace } from "../models/axis-model"
+
+import "./graph-drop-hint.scss"
+
+interface IProps {
+  hintText: string | undefined
+  targetW: number | undefined
+  targetH: number | undefined
+  place?: string | AxisPlace | undefined
+}
+
+export const GraphDropHint = ({ hintText, targetW, targetH, place }: IProps) => {
+  const hintDiv = useRef<HTMLDivElement>(null)
+  const [hintFoundW, setHintFoundW] = useState(0);
+  const [hintFoundH, setHintFoundH] = useState(0);
+
+  useLayoutEffect(() => {
+    if (hintDiv.current){
+      setHintFoundW(hintDiv.current?.offsetWidth)
+      setHintFoundH(hintDiv.current?.offsetHeight)
+    }
+  }, [hintDiv]);
+
+  function getLeftVal(wrapW: number, hintW: number){
+    return (wrapW * .5) - (hintW * .25)
+  }
+
+
+  function getTopVal(wrapH: number, hintH: number){
+    if (place && place === "bottom") {
+      return 300 // TODO replace w/ a calc based on graph height or mouse position
```

Comment:
I will either need to use the mouse position - looks like V2 does that - or bring in graphHeight as a prop and calculate based on that for all cases.

Reply:
Mouse position is preferred since the axis can be quite wide and we don't want the hint appearing far from where the user's attention is focussed.

Reply:
OK, perfect

Reply:
ended up using starting coordinates + deltas to get mouse position and set hint